app/validators/list.ts

```diff
@@ -0,0 +1,49 @@
+import { ValidatorFunction, ValidatorResult } from 'ember-changeset-validations';
+
+export type ListValidatorFunction = (
+    key: string,
+    newValue: undefined | unknown,
+    oldValue: undefined | unknown,
+    changes: Record<string, unknown>,
+    content: Record<string, unknown>,
+) => true | ValidatorResult[];
+
+/**
+ * Validate each item in a list using the given validator.
+ */
+export default function validateList(validator: ValidatorFunction): ListValidatorFunction {
+    return (key, newValue, oldValue, changes, content) => {
+        if (typeof newValue === 'undefined') {
+            // This validator validates list elements.
+            // If there is no list, there's nothing to validate!
+            return true;
+        }
+        if (!Array.isArray(newValue) || (typeof oldValue !== 'undefined' && !Array.isArray(oldValue))) {
+            throw new Error('validateList can only be used for array properties');
+        }
+        const results: ValidatorResult[] = [];
+        newValue.forEach((newItem, index) => {
```

Comment:
This would feel more natural to me as a `map`.

Reply:
Definitely. I wonder why it's not? 🤔